### PR TITLE
refactor: convert mstrophy show.tsx to function

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Related to #57388

I do need to reseed my database because the Playwright test user wasn't signed in; but the page loads and everything appears to work in the browser.

### CODE REVIEW

File: client/src/templates/Challenges/ms-trophy/show.tsx [hunk 1]
```diff
@@ -82,16 +82,34 @@ interface MsTrophyProps {
...
+  } = props;
+  initTests(tests);
+  updateChallengeMeta({
+    ...challengeMeta,
+    title,
+    challengeType,
+    helpCategory
+  });
+  challengeMounted(challengeMeta.id);
+  container.current?.focus();
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/ms-trophy/show.tsx:111]:
```suggestion
```
There's no need to call these on every render. This should map to the original `componentDidMount` callback, i.e. get called once when the component is first created.

File: client/src/templates/Challenges/ms-trophy/show.tsx [hunk 2]
```diff
@@ -116,133 +134,99 @@ class MsTrophy extends Component<MsTrophyProps> {
       helpCategory
     });
     challengeMounted(challengeMeta.id);
-    this.container.current?.focus();
-  }
+    container.current?.focus();
+  }, [props]);
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/ms-trophy/show.tsx:138]:
```suggestion
    // This effect should be run once on mount
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for taking this on, Anna :rocket:

**Comment by Sembauke** [comment]
@a2937  Do you need some help?

**Comment by a2937** [comment]
I'm pretty sure I've got this.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: Thanks, Anna!

**Review by Sembauke — APPROVED** [review]

**Pull request merged**